ja/lessons/ecto/associations.md

```diff
@@ -0,0 +1,498 @@
+---
+version: 1.0.1
+title: アソシエーション
+---
+
+このセクションでは、スキーマ間のアソシエーションを定義して扱うためにEctoを使用する方法について学びます。
+
+{% include toc.html %}
+
+## セットアップ
+
+前回のレッスンに続いて `Example` というアプリを構築します。 [ここ](../basics) のセットアップを見るとすぐに思い出せます。
+
+## アソシエーションの種類
+
+スキーマ間で定義することができるアソシーションは3つあります。それらがどういうものか、そして各種類の関係をどのように実装するのかを見ていきます。
+
+### Belongs To/Has Many
```

Comment:
I just didn't come up with a good way to translate these words into appropriate Japanese words. 
Plus, I thought it could be confusing if we translate them because Ecto uses the same keywords like "has_many" or "belongs_to" as macros.

Let me know if you have a better idea!

Reply:
たしかに `1対1` などの翻訳も考えていて、その場合 `1対多` と `1対1` の場合がある `Belongs to` はどうしようかなと思っていたのですが、たしかにここで表現しているのはあくまでもアソシエーションの性質ですね。このアソシエーションを用いる実際のデータ構造を考慮するのはあまり意味がないですね！
ということで、これらは `1対1` などと翻訳します！

Reply:
このように翻訳してみました。

- Belongs to -> 従属
- Has one -> 1対1
- Has many -> 1対多
- Many to many -> 多対多

特に問題がなければこれでいいかと思いますが、混乱のもとになるようだったら英語に戻します！
(Ruby on railsのActive Recordはどう訳してるんだろうと思って見てみたら、Active Recordは [英語を使ってましたね](https://railsguides.jp/association_basics.html#%E9%96%A2%E9%80%A3%E4%BB%98%E3%81%91%E3%81%AE%E7%A8%AE%E9%A1%9E) 🤔  )